src/cmap/auth/mongodb_oidc/cache.ts

```diff
@@ -1,7 +1,48 @@
+/* 5 minutes in milliseconds */
+const EXPIRATION_BUFFER_MS = 300000;
+
+/**
+ * An entry in a cache that can expire in a certain amount of time.
+ */
+export abstract class ExpiringCacheEntry {
+  expiration: number;
+
+  /**
+   * Create a new expiring token entry.
+   */
+  constructor(expiration: number) {
+    this.expiration = this.expirationTime(expiration);
+  }
+  /**
+   * The entry is still valid if the expiration is more than
+   * 5 minutes from the expiration time.
+   */
+  isValid() {
+    return this.expiration - Date.now() > EXPIRATION_BUFFER_MS;
+  }
+
+  /**
+   * Get an expiration time in milliseconds past epoch. Defaults to immediate.
+   */
+  private expirationTime(expiresInSeconds: number): number {
+    return Date.now() + expiresInSeconds * 1000;
+  }
+}
+
+/**
+ * An OIDC token cache.
+ */
+export interface Cache {
+  /**
+   * Implement the cache key for the token.
+   */
+  cacheKey(address: string, username: string, callbackHash: string): string;
+}
```

Comment:
Seems like the abstract class already covers enforcing the cacheKey method exists on all subclasses, what is this providing?